test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs

```diff
@@ -308,6 +309,43 @@ public async Task FluentMockServer_Proxy_Should_preserve_cookie_header_in_proxie
             Check.That(receivedRequest.Cookies).ContainsPair("name", "value");
         }
 
+        /// <summary>
```

Comment:
can you please also add a more low-level unit-test to test just this class with the changed logic for UTF8 ?
See https://github.com/WireMock-Net/WireMock.Net/blob/master/test/WireMock.Net.Tests/Util/BodyParserTests.cs for the tests.